sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -362,7 +620,7 @@ object Hash {
     final override def addTypedValue(value: Value): this.type =
       value match {
         case Value.ValueUnit =>
-          addByte(0.toByte)
+          addByte(0.toByte, "0 (unit)")
         case Value.ValueBool(b) =>
           addBool(b)
         case Value.ValueInt64(v) =>
```

Comment:
This could be a nice enhancement (completely optional)

```suggestion
    final override def addTypedValue(value: Value): this.type = {
      def context = value.productPrefix
      value match {
        case Value.ValueUnit =>
          addByte(0.toByte, context)
        case Value.ValueBool(b) =>
          addBool(b, context)
        case Value.ValueInt64(v) =>
```

Reply:
I'll add a comment for it as an improvement for later